### ISSUE TITLE
SQLCipher entering passphrase and encoded raw key not working

### ISSUE THREAD

**Issue by rbk0** [sqlitebrowser/sqlitebrowser#2908]
Details for the issue
--------------------

#### What did you do?
Hi, I was trying out the SQLCipher functionality of sqlitebrowser and had some weird behaviour.
I created a Test.db (attached with .env, default SQLCipher 4 values) with a raw key (0x61) and tried to open it using the - hexdecoded - passphrase (a).

[Test.zip](https://github.com/sqlitebrowser/sqlitebrowser/files/7576207/Test.zip)

I also tried to insert 0x00 before and after raw key until 64 characters, because from [Pragma Key Documentation](https://www.zetetic.net/sqlcipher/sqlcipher-api/#PRAGMA_key) I was not sure if the application is responsible of passing exactly 64 characters.

#### What did you expect to see?
As far as I understood it should be possible to either enter the rawkey or the password with sqlitebrowser(SQLCipher). Did I miss something?

#### What did you see instead?
Could not access Database.

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [x] Windows: ( _version:_ Windows 10 64bit )
- [ ] Linux: ( _distro:_ 11.2.1 )
- [x] macOS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [x] 3.12.2
- [ ] 3.12.1
- [ ] 3.12.0
- [ ] 3.11.x
- [ ] Other: ___

#### Did you also

- [] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by chrisjlocke** [comment]
When setting the encryption (or entering the key) did you set the dropdown to 'rawkey' ?

![image](https://user-images.githubusercontent.com/7872218/142771554-1d8d16c4-65c0-42ad-aebf-6fc74412574d.png)

**Comment by rbk0** [comment]
Yes, selection for passphrase/raw key was was definitely correct.  I tried GUI and .env file. Both same result.

And prefix 0x was also correct (not possible to miss that cause GUI requires it before entering raw key).

It is really easy to test.

1. Just create sample DB and encrypt using default. SQLCipher 4 settings.
2. Use some a simple passphrase (e.g. a)
3. Opening DB with raw key and enter encoded password (e.g. 0x61 in case of using "a" as passphrase).

**Comment by chrisjlocke** [comment]
I might be talking out of my bottom, so excuse me (I don't use sqlCipher much) but doing cursory searches, the raw key would be a 64-character hex string, not just a representation of the passphrase.  
Again, this post would indicate you have to rekey if you wish to switch from passphrase to raw key.
https://discuss.zetetic.net/t/moving-from-passphrase-to-raw-key-format/4047
I don't think you can simply 'intermingle', but again, I have no idea what I'm talking about, so may have misread.

**Comment by Petrakeas** [comment]
I am also facing this issue using 3.12.2

**Comment by elvary** [comment]
in my situation this sql file was exported from an unencrypted .sqlite file which is a record of brave browser history, unable to view its contents with db browser for sqlite because i have no idea what is the passphrase/raw key since i didnt create any when i exported it
![image](https://user-images.githubusercontent.com/69122770/188271646-e3933f92-d0ed-4847-a9a7-e1d72f468bb2.png)
![image](https://user-images.githubusercontent.com/69122770/188271666-13c4a22a-7f8f-427d-a8b1-24dcb354e551.png)
i tried opening the same .sql file with visual studio code and it displayed the contents normally without asking for encryption key, but vscode doesnt display contents in tables and columns like dbbrowserforsqlite so it's very intuitive to view;
so i guess there must be a default encryption key of this bd browser for sqlite, i just dont know what it is

**Comment by chrisjlocke** [comment]
You're mentioning two different types of files.  a .sqlite file is different from a .sql file.

Can you attach a screenshot of part of the file you can see in VSCode?  Blur any contents you don't want public, or crop, but I just want to ascertain what sort of file you're looking at.

Generally, DB4S will try and open the database without any key.  If it fails or doesn't see the file as a database (which a .sql file isn't) then it'll assume its encrypted and display the 'enter a passphrase' dialog.
